h/sentry/helpers/filters.py

```diff
@@ -29,3 +31,40 @@ def filter_ws4py_handshake_error(event):
         if str(event.exception) == "Header HTTP_UPGRADE is not defined":
             return False
     return True
+
+
+def filter_retryable_error(event):
+    """
+    Filter exceptions from requests that are going to be retried.
+
+    If a request raises a retryable error, so pyramid_retry automatically
+    retries that request, and a subsequent retry succeeds and we end up sending
+    a successful response back to the client, then we don't want to report
+    anything to Sentry.
+
+    See:
+
+    https://docs.pylonsproject.org/projects/pyramid-retry/en/latest/api.html#pyramid_retry.is_error_retryable
+    """
+    request = get_current_request()
+
+    if request is None:
+        # get_current_request() returns None if we're outside of Pyramid's
+        # request context. This happens when sentry-sdk's Pyramid integration
+        # catches an exception that Pyramid is going to raise to the WSGI
+        # server (Gunicorn, in our case).
+        # Always allow these uncaught exceptions to be reported to Sentry:
+        return True
+
+    if is_error_retryable(request, event.exception):
+        # Don't report the exception to Sentry if the request is going to be
+        # re-tried.
+        #
+        # Note: is_error_retryable() returns *False* if the error is marked as
```

Comment:
See [docs](https://hyp.is/2saE9OPeEemalOP0h6kAHw/docs.pylonsproject.org/projects/pyramid-retry/en/latest/api.html).